framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -3577,7 +3577,8 @@ public AnnotatedTypeMirror widenToUpperBound(
         final TypeMirror wildcardUBTypeMirror = wildcard.getExtendsBound().getUnderlyingType();
         if (types.isSubtype(wildcardUBTypeMirror, toModifyTypeMirror)) {
             return annotatedTypeMirror;
-        } else if (types.isSubtype(toModifyTypeMirror, wildcardUBTypeMirror)) {
+        } else if (types.isSubtype(
+                types.erasure(toModifyTypeMirror), types.erasure(wildcardUBTypeMirror))) {
```

Comment:
Can you add a comment why the types are erased in this case but not in the one above?